### PR TITLE
Fix the thread safe issue of caches manager and specify the default caches array value.

### PR DESCRIPTION
**Pull request by dreampiggy**
Update the test cases and documents about this behavior.

### New Pull Request Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar pull request in the [project](https://github.com/rs/SDWebImage/pulls) and found none

* [x] I have updated this branch with the latest master to avoid conflicts (via merge from master or rebase)
* [x] I have added the required tests to prove the fix/feature I am adding
* [x] I have updated the documentation (if necessary)
* [x] I have run the tests and they pass
* [x] I have run the lint and it passes (`pod lib lint`)

This merge request fixes / refers to the following issues: #841 

### Pull Request Description

During my detailed reviewing of 5.0's new feature, I found this `SDImageCachesManager`, does not keep thread-safe unlike `SDImageLoadersManager`. And it does not apply a default value unlike `SDImageLoadersManager`.

These two `SDImageCachesManager` and `SDImageLoadersManager` are the similar class, to support multiple cacehs and multiple downloaders. They should have the same design and concept. So this fix it.

Does not effect the API, only the implementation.

### CODE REVIEW

File: SDWebImage/SDImageCachesManager.m [hunk 1]
```diff
@@ -103,7 +119,9 @@ - (void)storeImage:(UIImage *)image imageData:(NSData *)imageData forKey:(NSStri
     if (!key) {
         return;
     }
-    NSArray<id<SDImageCache>> *caches = [self.caches copy];
+    SD_LOCK(self.cachesLock);
+    NSArray<id<SDImageCache>> *caches = self.caches;
```
Comment (zhongwuzw) [inline SDWebImage/SDImageCachesManager.m]: @dreampiggy @kinarob This is not thread-safe. 🤔

File: SDWebImage/SDImageCachesManager.h [hunk 2]
```diff
@@ -58,7 +58,7 @@ typedef NS_ENUM(NSUInteger, SDImageCachesManagerOperationPolicy) {
 /**
  All caches in caches manager. The caches array is a priority queue, which means the later added cache will have the highest priority
  */
-@property (atomic, copy, readwrite, nullable) NSArray<id<SDImageCache>> *caches;
+@property (nonatomic, copy, readwrite, nullable) NSArray<id<SDImageCache>> *caches;
```
Comment (zhongwuzw) [inline SDWebImage/SDImageCachesManager.h]: The same.

File: SDWebImage/SDImageCachesManager.m (hunk 1 above)
Comment (dreampiggy) [reply SDWebImage/SDImageCachesManager.m]: What means, not thread-safe ?

It's rare that people modify this value. It's designed to be a setup logic in `AppDelegate.m` actually.

Previous version, see: SDImageLoadersManager, using a mutable array, each method call need to `copy` a instance. This one does not.

Comment (dreampiggy) [reply SDWebImage/SDImageCachesManager.m]: This code actually already in master branch from v4.4.0.

See #2274

**Comment by zhongwuzw** [comment]
I'll fix this.

**Pull request merged**